app/javascript/packs/admin/config/smtp.js

```diff
@@ -0,0 +1,25 @@
+const checkbox = document.getElementById('settings_smtp_own_email_server');
+
+if (checkbox) {
+  checkbox.addEventListener('change', function () {
+    const customSMTPSection = document.getElementsByClassName(
+      'js-custom-smtp-section',
+    )[0];
+
+    if (this.checked) {
+      customSMTPSection.classList.remove('hidden');
+    } else {
+      // when the user indicates that they do not want to use their own server
+      // we clear the form values except for those fields that have a default value.
+      const inputs = customSMTPSection.getElementsByTagName('input');
+      const haveDefaultValues = ['authentication'];
+
+      for (let i = 0; i < inputs.length; i++) {
+        if (!haveDefaultValues.some((el) => inputs.item(i).name.includes(el))) {
+          inputs.item(i).value = '';
+        }
+      }
+      customSMTPSection.classList.add('hidden');
+    }
+  });
+}
```

Comment:
Refactor (blocking): Use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) instead as well as `event.target` instead of `this`.

```suggestion
document
  .getElementById('settings_smtp_own_email_server')
  ?.addEventListener('change', (event) => {
    const customSMTPSection = document.getElementsByClassName(
      'js-custom-smtp-section',
    )[0];

    if (event.target.checked) {
      customSMTPSection.classList.remove('hidden');
    } else {
      // when the user indicates that they do not want to use their own server
      // we clear the form values except for those fields that have a default value.
      const inputs = customSMTPSection.getElementsByTagName('input');
      const haveDefaultValues = ['authentication'];

      for (let i = 0; i < inputs.length; i++) {
        if (!haveDefaultValues.some((el) => inputs.item(i).name.includes(el))) {
          inputs.item(i).value = '';
        }
      }
      customSMTPSection.classList.add('hidden');
    }
  });

```

Refactor (nitpick): Prefer restructuring

```diff
-   const customSMTPSection = document.getElementsByClassName(
+   const [ customSMTPSection ] = document.getElementsByClassName(
      'js-custom-smtp-section',
-   )[0];
+   );
```

Refactor (nitpick): Prefer `for`/`of`

```diff
- for (let i = 0; i < inputs.length; i++) {
+ for (const input of inputs) {
- if (!haveDefaultValues.some((el) => inputs.item(i).name.includes(el))) {
+ if (!haveDefaultValues.some((el) => input.name.includes(el))) {
-    inputs.item(i).value = '';
+    input.value = '';
  }
}
```

Reply:
Thanks Nick,I'll update these!